### PR TITLE
Added support for adding solution items within a different folder name

### PR DESCRIPTION
**Pull request by icnocop**

### CODE REVIEW

File: src/Microsoft.VisualStudio.SlnGen.UnitTests/TempDirectory.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Corporation.
+//
+// Licensed under the MIT license.
+
+using System;
+using System.IO;
+
+namespace Microsoft.VisualStudio.SlnGen.UnitTests
+{
+    internal class TempDirectory : IDisposable
```
Comment (jeffkl) [inline src/Microsoft.VisualStudio.SlnGen.UnitTests/TempDirectory.cs:10]: The test base class `TestBase` already has an implementation to create a temp folder for each test.  You can access the `TestRootPath` folder to use it.

https://github.com/microsoft/slngen/blob/main/src/Microsoft.VisualStudio.SlnGen.UnitTests/TestBase.cs#L32

File: src/Microsoft.VisualStudio.SlnGen/SlnHierarchy.cs
Comment (jeffkl) [inline src/Microsoft.VisualStudio.SlnGen/SlnHierarchy.cs:1]: Where does a user specify the name for a solution folder?

Comment (icnocop) [reply src/Microsoft.VisualStudio.SlnGen/SlnHierarchy.cs:1]: The first parameter to the new, overloaded, `AddSolutionItems` method.

File: src/Microsoft.VisualStudio.SlnGen.UnitTests/TempDirectory.cs (hunk 1 above)
Comment (icnocop) [reply src/Microsoft.VisualStudio.SlnGen.UnitTests/TempDirectory.cs]: Thank you. I've pushed another commit to resolve this.

**Review by jeffkl — APPROVED** [review]
Thank you for the contribution!

**Pull request merged**